typescript/packages/subsurface-viewer/src/layers/wells/utils/spline.ts

```diff
@@ -246,14 +315,29 @@ export function coarsenWells(data_in: FeatureCollection): FeatureCollection {
             continue;
         }
 
-        const options = {
-            tolerance: 0.01,
-            highQuality: false,
-            mutate: false,
-        };
-
-        const coordsSimplified = simplify(lineString, options);
-        lineString.coordinates = coordsSimplified.coordinates as Position3D[];
+        const n = lineString.coordinates.length;
+        const isHorizontalWell =
+            lineString.coordinates[0][0] == lineString.coordinates[n - 1][0] &&
+            lineString.coordinates[0][1] == lineString.coordinates[n - 1][1];
```

Comment:
This doesn't look like a proper check for a horizontal or vertical well

Reply:
I think this is sufficient in 99.9999% of the cases. But I could of course di a more expensive test and use all points in the well path.